### PR TITLE
feat(protocol-designer): Add base thermocycler form

### LINKED ISSUE #5566: TC Step Form (Blank)
As a PD user, I would like to be able to fill out a basic state change step for the Thermocycler

## Acceptance Criteria
- [ ] New “Thermocycler” step type in “Add Step” dropdown
- [ ] Clicking “Thermocycler” step opens a new “Thermocycler” step form
- [ ] Step form has one section titled “Thermocycler”
- [ ] Step form has radio buttons to select Thermocycler step type.
  - [ ] “Change thermocycler state”
  - [ ] “Program a thermal cycling profile”
  - [ ] Neither selected by default
  - [ ] “Program a thermal cycling profile” is disabled (for now)
- [ ] Step form has buttons for “Delete” and “Notes”
- [ ] User can click “Close” to close the form without saving data

## Design
- [Add Step Dropdown (Invision)](https://projects.invisionapp.com/share/VUWZT27EDJ8#/screens/414817321)
- [Step Type Toggle (Invision)](https://projects.invisionapp.com/share/VUWZT27EDJ8#/screens/414817324)

### PR DESCRIPTION
**Pull request by Kadee80**
## overview

This PR closes #5566 by adding the initial 2 form fields for thermocycler form with the profile step disabled. It also cleans up some older logic that allowed temperature steps with a TC on the deck, and removes TC from temperature step and pause step dropdowns.

## changelog

- feat(protocol-designer): Add base thermocycler form
- refactor: Remove TC logic from pause and temperature steps

## review requests

- Make a protocol with a TC but no temperature module
- Add a TC step
- [ ] Initial form renders
- [ ] Profile step is disabled
- [ ] Temperature step is disabled in the step creation button
- Make a pause step
- [ ] Pause until temperature reached option is disabled and the tooltip only references the temperature module
- Add a temperature module to the deck
- [ ] Temperature step creation button is enabled
- [ ] Only temperature module shows up in module dropdown (Follow up PR question: do we want to just make this text instead of a dropdown like in magnet step?)
- Add a pause step
- [ ] Pause until temp option is enabled
- [ ] Only temperature module shows up in module dropdown

## risk assessment
Low. TC in PD behind a FF.

### CODE REVIEW

**Review by IanLondon — APPROVED** [review]
:rainbow: good catch on the old TEMPERATURE stuff

File: protocol-designer/src/components/StepEditForm/forms/ThermocyclerForm/index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import { i18n } from '../../../../localization'
+
+import { RadioGroupField } from '../../fields'
+import styles from '../../StepEditForm.css'
+
+import type { FormData } from '../../../../form-types'
+import type { FocusHandlers } from '../../types'
+
+type TCFormProps = { focusHandlers: FocusHandlers, formData: FormData }
+
```
Comment (shlokamin) [inline protocol-designer/src/components/StepEditForm/forms/ThermocyclerForm/index.js:14]: exact props

**Review by shlokamin — CHANGES_REQUESTED** [review]
Lookin good! Great catch with the temp vs tc form stuff. A few things:

(1) Looks like there's a missing exact type check
(2) This doesn't seem to be working for me: 

> Make a pause step
 Pause until temperature reached option is disabled and the tooltip only references the temperature module

I'm still seeing the `Pause until temperature reached` option enabled, and I have no temperature module on the deck (see screenshot attached). 
Screen Shot 2020-05-07 at 10 39 16 AM

**Review by shlokamin — APPROVED** [review]
🚢

**Pull request merged**